src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/Reliability/SqlRetryStrategy.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+
+namespace Microsoft.Data.SqlClient.Reliability
+{
+
+    /// <summary>
+    /// Defines a callback delegate that will be invoked whenever a retry condition is encountered.
+    /// </summary>
+    /// <param name="retryCount">The current retry attempt count.</param>
+    /// <param name="lastException">The exception that caused the retry conditions to occur.</param>
+    /// <param name="delay">The delay that indicates how long the current thread will be suspended before the next iteration is invoked.</param>
+    /// <returns><see langword="true"/> if a retry is allowed; otherwise, <see langword="false"/>.</returns>
+    public delegate bool ShouldRetry(int retryCount, Exception lastException, out TimeSpan delay);
+
+    /// <summary>
+    /// Represents a retry strategy that determines the number of retry attempts and the interval between retries.
+    /// </summary>
+    public abstract class SqlRetryStrategy
+    {
+        #region Public members
+        /// <summary>
+        /// Represents the default number of retry attempts.
+        /// </summary>
+        public static readonly int DefaultClientRetryCount = 3;
+
+        /// <summary>
+        /// Represents the default amount of time used when calculating a random delta in the exponential delay between retries.
+        /// </summary>
+        public static readonly TimeSpan DefaultDeltaBackoff = TimeSpan.FromSeconds(20.0);
```

Comment:
These should not be fields but properties. There is no need to expose fields which are an implementation detail. A trivial property hides that but causes no performance impact.

Also, I wonder if these values should be exposed at all. They are very application-specific. Also, the very concept of backoff and "increment" is specific to the concrete `SqlRetryStrategy`. Some strategies would not understand this very notion. It seems like a layering violation to put such values on the base class `SqlRetryStrategy`.

Reply:
Good point, will look into it.